Sources/TuistSupport/Extensions/String+Extras.swift

```diff
@@ -184,4 +163,42 @@ extension String {
     mutating func shellEscape() {
         self = shellEscaped()
     }
+
+    private func inShellWhitelist(_ codeUnit: UInt8) -> Bool {
+        switch codeUnit {
+        case UInt8(ascii: "a") ... UInt8(ascii: "z"),
+             UInt8(ascii: "A") ... UInt8(ascii: "Z"),
+             UInt8(ascii: "0") ... UInt8(ascii: "9"),
+             UInt8(ascii: "-"),
+             UInt8(ascii: "_"),
+             UInt8(ascii: "/"),
+             UInt8(ascii: ":"),
+             UInt8(ascii: "@"),
+             UInt8(ascii: "%"),
+             UInt8(ascii: "+"),
+             UInt8(ascii: "="),
+             UInt8(ascii: "."),
+             UInt8(ascii: ","):
+            return true
+        default:
+            return false
+        }
+    }
+}
+
+public extension Array where Element: CustomStringConvertible {
+    /// Returns a sentence listing the elements contained in the array.
+    /// ["Framework"] results in "Framework"
+    /// ["Framework", "App"] results in "Framework and App"
+    /// ["Framework", "App", "Tests"] results in "Framework, App, and Tests"
+    /// - Returns: <#description#>
+    func listed() -> String {
+        if self.count == 0 { return "" }
+        if self.count == 1 { return self.first!.description }
+        if self.count == 2 { return "\(self[0]) and \(self[1])" } else {
+            var elements = self
+            let last = elements.popLast()!
+            return "\(elements.map(\.description).joined(separator: ", ")), and \(last.description)"
+        }
```

Comment:
Nice this will be a handy extension to use.

Minor suggestion, `Foundation` has a built in `ListFormatter` that achieves the same results

e.g.

```swift
let formatter = ListFormatter()
formatter.string(from: ["a"]) // a
formatter.string(from: ["a", "b"]) // a and b
formatter.string(from: ["a", "b", "c"]) // a, b and c
```